public/app/features/explore/state/datasource.ts

```diff
@@ -44,6 +45,12 @@ export function changeDatasource(
     const { history, instance } = await loadAndInitDatasource(orgId, { uid: datasourceUid });
     const currentDataSourceInstance = getState().explore[exploreId]!.datasourceInstance;
 
+    if (instance.meta.mixed) {
+      reportInteraction('explore_change_ds_to_mixed');
+    } else if (currentDataSourceInstance?.meta.mixed) {
+      reportInteraction('explore_change_ds_from_mixed');
+    }
+
```

Comment:
given we are making this change i think it's worth to get as much data as possible. wdyt?
```suggestion
    reportInteraction('explore_change_ds', {
      from: (currentDataSourceInstance?.meta?.mixed ? 'mixed' : currentDataSourceInstance?.type) || 'unknown',
      to: instance.meta.mixed ? 'mixed' : instance.type,
      exploreId,
    });
```